### DISCUSSION TITLE
Multithreaded optimisation in TASO

### DISCUSSION THREAD

**Discussion by lmondada** [CQCL/tket2#164] [General]
Multithreading in TASO:
- Parallel pattern matching with single queue (status: implemented)
- Circuit splitting and merging (status: #149)
- Circuit sharding: dispatch to process based on hash prefix (status: not implemented)

The first approach currently does not work well. Option 2 is promising and soon to be tested. Option 3 is at the idea stage.

This discussion is meant as a place to coordinate all threading-related tasks and improvement ideas for TASO. For the beta launch we need to target 8 cores.

**Comment by aborgna-q** [comment]
Options 2 and 3 are now implemented in ~TASO~ Badger.
